src/main/java/org/gridsuite/securityanalysis/server/entities/ComputationStatusEntity.java

```diff
@@ -4,34 +4,47 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-package org.gridsuite.securityanalysis.server.repository;
+package org.gridsuite.securityanalysis.server.entities;
 
-import lombok.AllArgsConstructor;
-import lombok.Getter;
-import lombok.Setter;
-import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
-import org.springframework.data.cassandra.core.mapping.Column;
-import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
-import org.springframework.data.cassandra.core.mapping.Table;
+import java.util.UUID;
 
 import java.io.Serializable;
-import java.util.UUID;
+
+import javax.persistence.Embeddable;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.Table;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
 
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 @Getter
-@Setter
 @AllArgsConstructor
-@Table("computationStatus")
+@NoArgsConstructor
+@Entity
+@IdClass(ComputationStatusEntity.ID.class)
+@Table(name = "computationStatus")
 public class ComputationStatusEntity implements Serializable {
 
-    @PrimaryKeyColumn(name = "resultUuid", type = PrimaryKeyType.PARTITIONED)
+    @NoArgsConstructor
+    @Getter
+    @Embeddable
+    static class ID implements Serializable {
+        private UUID resultUuid;
+
+        private String contingencyId;
+    }
+
+    @Id
     private UUID resultUuid;
 
-    @PrimaryKeyColumn(name = "contingencyId", type = PrimaryKeyType.CLUSTERED)
+    @Id
     private String contingencyId;
```

Comment:
I had the hesitation, because of the repetition, but went to IdClass, as less visible as possible from outside the persistence implementation. Which has the benefit of not modifying the tests, opening less space for tests deviations.